### ISSUE TITLE
Quarto Revealjs, switching from source to visual changes code

### ISSUE THREAD

**Issue by KrisKarul** [rstudio/rstudio#15322]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2024.09.0 Build 375
    OS Version      : Windows 11 Home
    R Version       : R version 4.3.1 (2023-06-16 ucrt)

### Steps to reproduce the problem

Here's my code

```
<table class="design-table1">
    <thead>
        <th>Provisioning</th>
        <th>Regulating</th>
        <th>Cultural</th>
        <th>Supporting</th>
    </thead>
    <tbody>
        <td>Include the supply of:
            <li>Food</li><li>Water</li><li>Raw materials</li></td>
        <td>Examples include:
          <li>Climate regulation</li><li>Flood control</li><li>Disease regulation</li><li>Waste management</li></td>
        <td>Examples include:
            <li>Recreational (ex: ecotourism)</li><li>Spiritual</li><li>Educational benefits</li></td>
        <td>Essential for other ES such as:
            <li>Nutrient cycling</li><li>Soil formation</li></td>
    </tbody>
</table>
```

### Describe the problem in detail

Currently working on a Quarto revealjs slides using RStudio. Whenever I switch from source to visual, the code changes, to the point that the output is different.

Here's one example where I use the <table> tag for a custom table appearance. Switching to visual changes it into a Markdown table format, which changes the table appearance as well.

![Quarto_1](https://github.com/user-attachments/assets/2d3224b2-57f2-439a-a144-602a63ddbf70)

### Describe the behavior you expected

Switching between source and visual should not change the code, as in do not render it to markdown

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@KrisKarul Thank you for raising this! The changes are made by Pandoc to conform to its preferred format. However, I think that the upcoming 2024.09.1 [Cranberry Hibiscus patch](https://dailies.rstudio.com/rstudio/cranberry-hibiscus/) will have what you're looking for. It's available at https://dailies.rstudio.com/rstudio/cranberry-hibiscus/ if you'd like to try it, but it's currently in development and for testing purposes only.

1. Open the file in visual mode.
2. <kbd>Format -> Raw -> HTML Block</kbd>
3. Paste the code in the raw block.

Now switching back-and-forth should keep the HTML code intact, and it'll be used when rendering the document.

**Comment by KrisKarul** [comment]
> 1. Open the file in visual mode.
> 2. Format -> Raw -> HTML Block
> 3. Paste the code in the raw block.
> 
> Now switching back-and-forth should keep the HTML code intact, and it'll be used when rendering the document.

This is for the upcoming [Cranberry Hibiscus patch](https://dailies.rstudio.com/rstudio/cranberry-hibiscus/) right?

**Comment by ronblum** [comment]
Correct. It's covered in issue #15189. A change to the latest Pandoc is interfering with the intended behavior. The Pandoc change was intentional, not a bug, so we've adapted RStudio to it.

**Comment by ronblum** [comment]
To clarify a bit, and why I didn't mark this as a duplicate:
- The Visual Editor uses Pandoc, which converts the contents to its preferred format. So what you're seeing--HTML tables switching to Markdown format--is expected behavior.
- Placing the HTML table in a raw HTML block (`{=html}`) will preserve it.
- There was a recent change to Pandoc which caused this not to work (#15189) in the most recent release of RStudio, 2023.09.0+375. We fixed RStudio to handle this in the upcoming patch.
